src/main/java/com/eprosima/fastdds/idl/templates/TypeObjectTestingTestSource.stg

```diff
@@ -630,15 +710,15 @@ check_struct_member(member, parent) ::= <<
     $get_type_identifier(type=member.typecode, var="member_type_ids")$
     if (TK_NONE != member_type_ids.type_identifier2()._d())
     {
-        EXPECT_TRUE(member_type_ids.type_identifier1() == type_objects.minimal_type_object.minimal().struct_type().member_seq()[pos].common().member_type_id() ||
-            member_type_ids.type_identifier1() == type_objects.complete_type_object.complete().struct_type().member_seq()[pos].common().member_type_id());
-        EXPECT_TRUE(member_type_ids.type_identifier2() == type_objects.minimal_type_object.minimal().struct_type().member_seq()[pos].common().member_type_id() ||
-            member_type_ids.type_identifier2() == type_objects.complete_type_object.complete().struct_type().member_seq()[pos].common().member_type_id());
+        EXPECT_TRUE(compare_type_identifiers(member_type_ids.type_identifier1(), type_objects.minimal_type_object.minimal().struct_type().member_seq()[pos].common().member_type_id()) ||
+                compare_type_identifiers(member_type_ids.type_identifier1(), type_objects.complete_type_object.complete().struct_type().member_seq()[pos].common().member_type_id()));
+        EXPECT_TRUE(compare_type_identifiers(member_type_ids.type_identifier2(), type_objects.minimal_type_object.minimal().struct_type().member_seq()[pos].common().member_type_id()) ||
+                compare_type_identifiers(member_type_ids.type_identifier2(), type_objects.complete_type_object.complete().struct_type().member_seq()[pos].common().member_type_id()));
```

Comment:
The same issue is probably going to be happening with union members. Would you mind adding a test case and checking?

Reply:
added in [1992c80](https://github.com/eProsima/Fast-DDS-Gen/pull/297/commits/1992c8069acc41c3e00f3f4232adc7bc8d482d84)